### PR TITLE
Added IPCodes, international protection markings

### PR DESCRIPTION
**Pull request by mattr555**
Hopefully I did this right! :laughing:
Continued from duckduckgo/zeroclickinfo-goodies#1139

**What does your Instant Answer do?**
Decodes IP Code markings

**What problem does your Instant Answer solve (Why is it better than organic links)?**
These are a standard and simple for answers.

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://en.wikipedia.org/wiki/IP_Code

**Why did you choose this data source?**
It's correct

**What are some example queries that trigger this Instant Answer?**
`IP68`, `ip 42`

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
People working with machinery/parts that need to be protected from particle ingress (like roboticists)

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
[My own.](https://duck.co/ideas/idea/5227/ip-international-protection-marking-codes)

**Which existing Instant Answers will this one supercede/overlap with?**
None

**Are you having any problems? Do you need our help with anything?**
All good, I just want to see what it looks like though :)

## Checklist
Please place an 'X' where appropriate.

```
[x] Added metadata and attribution information
[x] Verified that Instant Answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[x] Verified that Instant Answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility
    Please let us know which browsers/devices you've tested on:
    I can't :(
```

------
IA Page: http://duck.co/ia/view/ip_codes

### CODE REVIEW

**Comment by moollaza** [comment]
@mattr555 thanks! We'll be sure to review this shortly and give you some feedback!

**Comment by mintsoft** [comment]
@moollaza I guess there's no way for the community to test this atm?

**Comment by moollaza** [comment]
@mintsoft unfortunately, not right now.

**Comment by jdorweiler** [comment]
@mattr555 Thanks!  I'll get this on a test machine ASAP.

**Comment by jdorweiler** [comment]
@mattr555 what trigger words would you like?  
Could you rename your directory to fathead/ip_codes too?

File: share/fathead/i_p_codes/parse.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+if __name__ == "__main__":
+    output = []
+    for (part_num, part_desc), (water_num, water_desc) in product(particles.items(), water.items()):
+        name = "IP{}{}".format(part_num, water_num)
+        desc = "Particle protection: {}<br>Water protection: {}".format(part_desc, water_desc)
+        output.append(article(name, desc))
+        output.append(redirect('IP {}{}'.format(part_num, water_num), name))
+
+    with open('output.txt', 'wb') as f:
+        writer = csv.writer(f, delimiter='\t')
```
Comment (jdorweiler) [inline share/fathead/i_p_codes/parse.py]: I'm having some trouble with the output files.  IIRC I ran into problems with a PR a while ago and I think csv.writer was the cause.

**Comment by mattr555** [comment]
@jdorweiler Let me know if printing to standard out and piping to a file works better.

Is it possible to do this one without trigger words? If not, could we do a word of "code" at the end (i.e. `IP67 code`)

**Comment by jdorweiler** [comment]
@mattr555  Let me know what you think.  It works without a trigger word but I added `code, codes` as triggers just in case. 
https://ddh10.duckduckgo.com/?q=ip49k&ia=about

**Comment by jdorweiler** [comment]
I'll change the moreAt link to Wikipedia.

**Comment by mintsoft** [comment]
@jdorweiler @mattr555 I don't think it's quite triggering right:
* https://ddh10.duckduckgo.com/?q=ip55
* https://ddh10.duckduckgo.com/?q=ip65

**Comment by jdorweiler** [comment]
@mintsoft I noticed that the wiki article does show up for some of them.  Adding a space seems to work though.  https://ddh10.duckduckgo.com/?q=ip+65&ia=about

**Comment by mintsoft** [comment]
@jdorweiler I see, however shouldn't this still trigger: https://ddh10.duckduckgo.com/?q=ip+55&ia=about

**Comment by jdorweiler** [comment]
@mintsoft That's where the trigger words come in.  Without it the wiki article will take precedence.
https://ddh10.duckduckgo.com/?q=ip+55+code&ia=about

**Comment by mintsoft** [comment]
Ah right, fair enough then!

**Comment by jdorweiler** [comment]
@mattr555 ready to go!  I'll ping you when it's live.

**Comment by jdorweiler** [comment]
@mattr555 It's live! https://duckduckgo.com/?q=ip+55+code&ia=about  Sorry for the long delay deploy it.

**Comment by mattr555** [comment]
@jdorweiler :+1: Thanks for the help!

**Pull request merged**